src/pages/Admin/Core/Templates/index-fund/IndexFundOwner/useUpdateOwner.ts

```diff
@@ -35,6 +36,13 @@ export default function useUpdateOwner() {
         children: "Wallet is not connected",
       });
 
+    if (!isPolygonChain(wallet.chain.chain_id)) {
```

Comment:
since this check is repeated on every template, it's better to bring this up 
e.g in `Admin/Guard`
```
<context.Provider value={data & { walletOrPrompt }>
```

and use per template
```ts 
const {walletOrPrompt} = useAdminResources()
//some type guard
if(walletOrPrompt is prompt) return prompt()

/** a valid wallet */

```